core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -226,7 +226,11 @@ object SparkSubmit {
 
     // Set the cluster manager
     val clusterManager: Int = args.master match {
-      case m if m.startsWith("yarn") => YARN
+      case "yarn" => YARN
+      case "yarn-client" | "yarn-cluster" =>
```

Comment:
maybe something like `Master must either be yarn or start with spark, mesos or local`